docs/Additional-Activation-OTP.md

```diff
@@ -0,0 +1,168 @@
+
+   1. The user scans QR code or retypes the activation code manually.
+   1. Mobile application then initializes the activation, using activation code.
+
+1. Intermediate Server Application receives a regular activation request, with activation code:
+   
+   1. The activation code is verified by the PowerAuth server in the [`prepareActivation`](https://github.com/wultra/powerauth-server/blob/develop/docs/SOAP-Service-Methods.md#method-prepareactivation) method. 
+   1. If everything is right, the activation is set to the OTP_USED state.
+   1. At the same time, Intermediate Server Application initiates the delivery of activation OTP. It's recommended to deliver such code via the dedicated out-of-band channel, for example, via SMS.
+
+1. The mobile application receives the response from the server and completes the keys-exchange on the mobile side.
+   
+Now it depends whether the user has to retype OTP back to the Master Front-End Application, or the mobile application.
+ 
+1. For the first case, the implementation is straightforward. Once the user retypes OTP back to Master Front-End Application, the activation can be completed on PowerAuth Server by calling [`commitActivation`](https://github.com/wultra/powerauth-server/blob/develop/docs/SOAP-Service-Methods.md#method-commitactivation) method. In case the commit fails, the number of commit attempts is limited to the [`MAX_FAILED_ATTEMPTS`](Computing-and-Validating-Signatures.md#constants-and-variables).
+
+2. In case the OTP is retyped in the mobile application, the additional RESTful endpoint has to be implemented on the Intermediate Server Application. We recommend to use our [ECIES encryption](End-To-End-Encryption.md) to protect such endpoint. In case the commit fails, the number of commit attempts is limited to the [`MAX_FAILED_ATTEMPTS`](Computing-and-Validating-Signatures.md#constants-and-variables). 
+
+For both cases, it's recommended to generate a new OTP in case that delivery failed (e.g. user did not receive SMS). You can use [`updateActivationOtp`](https://github.com/wultra/powerauth-server/blob/develop/docs/SOAP-Service-Methods.md#method-updateactivationotp) method to set a new OTP to the PowerAuth Server.
+
+You can also slightly alter this whole sequence, and generate the first OTP later, in the step 3. In this case, you have to use [`updateActivationOtp`](https://github.com/wultra/powerauth-server/blob/develop/docs/SOAP-Service-Methods.md#method-updateactivationotp) in the same step, to set the OTP. 
+
+### Confirm custom activation with OTP
+
+There are multiple ways how to implement custom activation and confirm it with an additional authentication OTP so that we will discuss only one particular case. In this scenario, we don't involve the Master Front-End Application in the process, but we expect that the user has already issued some valid credentials to the system. 
+
+1. In the mobile application:
+   
+   1. The user enters login and password
```

Comment:
Not sure about the `login` word, it is better to use `username` or similar term.